### ISSUE TITLE
[TCK] TSAuthConfigFactory assertion even when not being used.

### ISSUE THREAD

**Issue by jeanouii** [jakartaee/authentication#117]
As a follow up of the previous ticket https://github.com/eclipse-ee4j/authentication/issues/116 

Again, from Jakarta Authentication user guide, TSAuthConfigFactory is not required to be set globally and could be registered into vendor's implementation of AuthConfigFactory.

If that is the case, I think there is a bug at https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/src/com/sun/ts/tests/jaspic/spi/servlet/Client.java#L2434

In this JASPIC Servlet test, there is no point if adding message assertion for TSAuthConfigFactory messages are we are not using it. 

Basically, TSAuthConfigFactory assertions should be specifically added when we are registering the factory ourselves for particular assertions. 

Other than that this is not what we are testing so we should not consider this particular factory to be used, right?

**Comment by scottmarlow** [comment]
@jeanouii Please do supply the required information mentioned in the TCK process document:

> Challenges MUST be filed via the specification project’s issue tracker using the label challenge and include the following information:
> 
>     The relevant specification version and section number(s)
> 
>     The coordinates of the challenged test(s)
> 
>     The exact TCK version
> 
>     The implementation being tested, including name and company
> 
>     A full description of why the test is invalid and what the correct behavior is believed to be
> 
>     Any supporting material; debug logs, test output, test logs, run scripts, etc.
>
